website/addons/s3/static/s3UserConfig.js

```diff
@@ -30,13 +30,52 @@ function ViewModel(url) {
     self.accounts = ko.observableArray();
 
     ChangeMessageMixin.call(self);
+    self.options = {
```

Comment:
I migrated existing s3-specific functionality into the generic class `OauthAddonNodeConfig`, which provides the folder picker UI and a set of automated tests for addons that use it. We eventually want all of them that can to use it for both ease of maintenance and UX consistency. While S3 isn't technically an `OauthAddon`, much of the back-end treats it as such -- our current naming convention may be a misnomer.

The user settings page has a generalized viewmodel for addons that *actually* support OAuth, but given that it worked with relatively few changes (really just changing how certain functions were accessed to allow the credentials modal to work), and that none of the tickets this was intended to solve were directly related to front-end maintenance, I thought writing another generalized viewmodel for addon_user_settings with modals that receive hard-coded credentials was out of scope of this PR. Possibly in scope of the [follow-follow-up ticket](https://openscience.atlassian.net/browse/OSF-6678), however